packages/zstd_static.rb

```diff
@@ -0,0 +1,1545 @@
+      xgc5+FpSvuOu5XtpP4qG5bP2wSnmsJ/zj/kzf+bP/Jk/82f+zJ/5M3/mz/yZP/Nn/syf+TN/5s/8mT/
+      zZ/78bp//B3xMHIIAGAEA
+      _eof
+      }
+
+      function buildall() {
+          [ -d "$BUILD_DIR/install" ] && rm -rf "$BUILD_DIR/install"
+          # setcross "$1"
+          setnative "$1"
+
+          DISABLE_WIRESHARK=""
+
+          if [ "$MUSL_CROSS" = "mips-linux-muslsf" ]; then
+              DISABLE_WIRESHARK="1"
+          fi
+
+          get_musl
+
+          ${CROSS_PFX}gcc $CFLAGS $LDFLAGS -o "$BIN_DIR/xxd" "$BUILD_DIR/xxd.c"
+          $STRIP "$BIN_DIR/xxd"
+
+          ${CROSS_PFX}gcc $CFLAGS $LDFLAGS -o "$BIN_DIR/sav" "$BUILD_DIR/sav.c"
+          $STRIP "$BIN_DIR/sav"
+
+          ${CROSS_PFX}gcc $CFLAGS $LDFLAGS -o "$BIN_DIR/pause" "$BUILD_DIR/pause.c" -nostartfiles
+          $STRIP "$BIN_DIR/pause"
+
+          build_xzdec
+
+          build_pv
+
+          build_zlib
+
+          build_xz
+          build_lz4
+          build_zstd
+      }
+
+      function main() {
+          HASH=$(sha256sum $0|head -c16)
+          mkdir -p "$BUILD_DIR"
+
+          if [ ! -f .gitignore ]; then
+              echo -e 'output\npatches\n.gitignore' > .gitignore
+          fi
+
+          if [ ! -d "$BUILD_DIR/patches" ]; then
+              unpack_patches "$BUILD_DIR"
+          fi
+
+          if [ ! -z "$1" ]; then
+              if [ "$1" = "unpack" ]; then
+                  true
+              elif [ "$1" = "repack" -a -d "$BUILD_DIR/patches" ]; then
+                  cp -a $0 $0.bak
+                  tar -C $BUILD_DIR -cz --owner=0 --group=0 patches xxd.c sav.c proberelay.c pause.c | base64 -w79 \
+                  | sed -e '/[<]_eof/,/^_eof/{//!d;r /dev/stdin' -e '}' $0 \
+                  | sponge $0
+              else
+                  buildall "$1"
+              fi
+              exit 0
+          fi
+
+          # armel-linux-musleabihf
+          # mips-linux-musl mips-linux-muslsf # mipsel-linux-musl mipsel-linux-muslsf mipsel-linux-musln32 mipsel-linux-musln32sf
+          for arch in armv7l-linux-musleabihf x86_64-linux-musl mips-linux-muslsf armv5l-linux-musleabihf aarch64-linux-musl
+          do
+              echo "** building for $arch **"
+              export arch
+              export PS4="+$arch+ "
+              buildall "$arch"
+              echo "** finished $arch bins **"
+          done
+
+          echo "** Finished **"
+      }
+
+      main "$@"
+      # vim:sw=4:ts=4:et:
+    BUILDER_EOF
+    case ARCH
+    when 'i686'
+      @cross = 'i686-linux-musl'
+    when 'x86_64'
+      @cross = 'x86_64-linux-musl'
+    when 'aarch64', 'armv7l'
+      @cross = 'arm-linux-musleabihf'
+    end
+    Kernel.system "bash ./crew-static-builder.sh #{@cross}"
+  end
+
+  def self.install
+    case ARCH
+    when 'i686'
+      @cross = 'i686-linux-musl'
+    when 'x86_64'
+      @cross = 'x86_64-linux-musl'
+    when 'aarch64', 'armv7l'
+      @cross = 'arm-linux-musleabihf'
+    end
```

Comment:
Why not add this logic outside all the class methods but still inside the class instead of duplicating in both build and install methods?

Reply:
Doesn't that mean that gets processed every time the file is read by crew? Otherwise it is only processed when a build or install happens.